### PR TITLE
Adding UI changes for plugin

### PR DESCRIPTION
**Pull request by Nepomuceno**
### What does this PR do?

Adding changes to the plugins UI and moving the spectrogram page.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you successfully ran tests with your changes locally? Including container validation.
* [x] Have you lint your code locally prior to submission?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you labelled this PR correctly? E.G. major, milestone, breaking, breaking-change, minor, feature, enhancement, patch, fix, chore, refactor etc

### CODE REVIEW

File: client/src/api/plugin/Queries.tsx [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 export function useGetPlugins() {
   return useQuery<PluginDefinition[]>(
     ['plugins'],
     async () => {
       const response = await axios.get<PluginDefinition[]>('/api/plugins').catch((err) => {
-        console.log('useGetPlugins error', err);
-        return { data: [] };
+        console.error('Failing getting plugins from api', err);
+        if (import.meta.env.IQENGINE_PLUGINS) {
+          const plugins = JSON.parse(import.meta.env.IQENGINE_PLUGINS);
```
Comment (Jaydee123) [inline client/src/api/plugin/Queries.tsx:12]: You probably want to check whether this parses correctly and not use this value if it doesn't, or give some indication it doesn't parse correctly

File: client/src/api/plugin/Queries.tsx [hunk 2]
```diff
@@ -17,3 +22,29 @@ export function useGetPlugins() {
...
 }
+
+export function useGetPlugin(plugin: PluginDefinition) {
+  return useQuery<string[]>(
+    ['plugin', plugin.name],
+    async () => {
+      const response = await axios.get<string[]>(plugin.url);
+      return response.data;
+    },
+    {
```
Comment (hannahlspencer) [inline client/src/api/plugin/Queries.tsx:33]: What would happen if the response was empty?

File: client/src/api/plugin/__tests__/Queries.test.tsx
Comment (hannahlspencer) [inline client/src/api/plugin/__tests__/Queries.test.tsx:1]: Should probably have tests for failure cases too

File: client/src/pages/spectrogram/SpectrogramPage.tsx [hunk 3]
```diff
@@ -305,15 +304,7 @@ export const SpectrogramPage = () => {
   };
 
   return (
-    <SpectrogramContext.Provider
```
Comment (hannahlspencer) [inline client/src/pages/spectrogram/SpectrogramPage.tsx:308]: Just a curiosity - how come we're not using the context component anymore?

Comment (Nepomuceno) [reply client/src/pages/spectrogram/SpectrogramPage.tsx:308]: It was never used by any component

File: client/src/api/plugin/__tests__/Queries.test.tsx
Comment (Nepomuceno) [reply client/src/api/plugin/__tests__/Queries.test.tsx:1]: Added 2 tests for failure scenarios

**Review by Jaydee123 — APPROVED** [review]

**Pull request merged**